### ISSUE TITLE
Issues when a component has additional "static" properties attached. 

### ISSUE THREAD

**Issue by ctrlplusb** [developit/preact-compat#223]
e.g.

``` typescript
function MyComponent(props) {
  const styles = MyComponent.styles;
  return <div className={css(styles.foo)}>Foo</div>;
}
MyComponent.styles = StyleSheet.create({
  foo: { color: 'red' }
});
```

**Comment by ctrlplusb** [comment]
I haven't dug into this deeply, but things settled the moment I changed to:

```typescript
const styles = StyleSheet.create({
  foo: { color: 'red' }
});
function MyComponent(props) {
  return <div className={css(styles.foo)}>Foo</div>;
}
```

Could possibly be have to do with the way files are bundled.

**Comment by developit** [comment]
Strange - I'm not sure what would be causing this...

**Comment by ctrlplusb** [comment]
Yeah, is a strange one. Maybe hold off any investigation - I will try and isolate the issue. Don't want to waste your time but put this up here just in case someone else experienced the same issue.

**Comment by developit** [comment]
Update: this will be fixed in Preact 8 as support for Pure Functional Components has been rewritten. It will go along with an update in `preact-compat` to support the change.

**Comment by ctrlplusb** [comment]
That is awesome news, and opens up some nice opportunities for me! 👏

**Comment by developit** [comment]
I can only assume this is fixed as of preact 8.0. LMK if I'm crazy!
